### DISCUSSION TITLE
Safeguards against SQL injections?

### DISCUSSION THREAD

**Discussion by GodGardensGuns** [supabase/supabase#440] [Questions] [answered]
I've been learning about SQL databases and managing DB security in production. I'm curious about what safeguards Supabase employs against SQL injections (if any). I know that, while using [Prisma](https://www.prisma.io/) to query an API, you can structure your queries to escape any SQL commands that are entered client side. Does Supabase/supabase-js have similar functionality?

**Comment by inian** [comment, marked as answer]
Hi @dayvista, we do a few things to protect against SQL Injections. Our client libraries encode each parameter as a [URL.SearchParam](https://developer.mozilla.org/en-US/docs/Web/API/URL/searchParams) which prevents you from injecting new parameters ([example](https://github.com/supabase/postgrest-js/blob/master/src/lib/PostgrestFilterBuilder.ts#L373)). 

The client library calls Postgrest which also [santizes strings](https://github.com/PostgREST/postgrest/issues/1591#issuecomment-696814970). 

If you do find a new attack vector, do email us at security@supabase.io
